lib/open_mic.rb

```diff
@@ -0,0 +1,39 @@
+class OpenMic
+  attr_reader :attr, :location, :date, :performers, :comp
+
+  def initialize(attr = {})
+    @attr = attr
+    @location = attr[:location]
+    @date = attr[:date]
+    @performers = []
+    @comp = {}
```

Comment:
Not sure that you _need_ this instance variable - see my notes in you all_jokes method below.